### PR TITLE
Default http.query() to urllib2, still allow requests

### PR DESCRIPTION
**Pull request by techhat**
This switches `salt.utils.http.query()` to use `urllib2`, with host name checking for SSL, but still allows `requests` to be used if desired.

### CODE REVIEW

**Comment by techhat** [comment]
Ping @rallytime. This needs to be thoroughly tested, especially using SSL with `urllib2`.

**Comment by techhat** [comment]
Bah, don't merge yet. This currently requires Python 2.7.9, which ships with `ssl.match_hostname`. Need to update to support `backports.ssl_match_hostname`.

**Comment by rallytime** [comment]
@techhat Should this be tested before it gets merged in? Or after?

**Comment by techhat** [comment]
Okay, backports version should now be supported.

@rallytime before merging, please.

**Comment by techhat** [comment]
Please don't merge yet. There may be other issues here.

File: salt/utils/http.py [hunk 1]
```diff
@@ -144,36 +180,85 @@ def query(url,
...
+            if not HAS_MATCHHOSTNAME:
+                log.warn(('match_hostname() not available, SSL hostname '
+                         'checking not available. THIS CONNECTION MAY NOT BE SECURE!'))
+            else:
+                hostname = request.get_host()
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                sock.connect((hostname, 443))
+                sockwrap = ssl.wrap_socket(
+                    sock,
+                    ca_certs='/etc/ssl/certs/ca-certificates.crt',
```
Comment (thatch45) [inline salt/utils/http.py]: This will be the real painful part, finding the cert. certiffi has a good example that we can start with

**Comment by techhat** [comment]
Close in favor of #19745.

**Pull request closed without merging**